src/App.tsx

```diff
@@ -1,14 +1,60 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = React.useState(true);
+  const [todos, setTodos] = React.useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = React.useState<Todo>();
+  const [completedFilter, setCompletedFilter] = React.useState<string | null>(
+    '',
```

Comment:
The `completedFilter` state is initialized with an empty string, but its type is `string | null`. Consider initializing it with `null` or ensuring that it is always a string to avoid potential type inconsistencies.